x-pack/plugins/reporting/server/lib/tasks/execute_report.ts

```diff
@@ -55,13 +57,15 @@ export class ExecuteReportTask implements ReportingTask {
   private kibanaId?: string;
   private kibanaName?: string;
   private store?: ReportingStore;
+  private getContentStream: ReturnType<typeof getContentStreamFactory>;
 
   constructor(
     private reporting: ReportingCore,
     private config: ReportingConfigType,
     logger: LevelLogger
   ) {
     this.logger = logger.clone(['runTask']);
+    this.getContentStream = getContentStreamFactory(reporting);
```

Comment:
`getContentStream` always seems to be called only once per file, so I am not seeing a benefit from having it returned from a factory function.

We could remove the factory wrapper by having `getContentStream` take `reporting` as a function argument. 

There are a lot of factory functions in Reporting, but it is a leftover of the old platform. The existing code that's like that should all be cleaned up in this PR: https://github.com/pulls